### ISSUE TITLE
Regexp rule with $domain modifier doesn't work in 4.4 extension when the length of a domain is specified

### ISSUE THREAD

**Issue by alexx7311** [AdguardTeam/tsurlfilter#136]
Steps to reproduce:
1. Disable all filters
2. Go to User rules and add `[$domain=/^webteizle\d{0,2}\.info/]##body`
3. Open the filtering log on a new page and go to https://webteizle.info/izle/altyazi/justice-league-crisis-on-infinite-earths-part-one-w1

The rule should work on webteizle.info.

**Comment by Alex-302** [comment]
Also not working
```
[$domain=/exampl[exyz]+\.com)/]##body
[$domain=/example\.(com|org)/]##body
```

![image](https://github.com/user-attachments/assets/93869ca8-188c-4d80-a1d7-7f118d144a8b)
![image](https://github.com/user-attachments/assets/6793b7e1-fc14-4159-a101-b6b8b94a5c06)

**Comment by scripthunter7** [comment]
Its a known problem; in the case of some modifiers, we need special tokenization in the future, but in the meantime the comma should be escaped, e.g.
```adblock
[$domain=/^webteizle\d{0\,2}\.info/]##body
```

**Comment by scripthunter7** [comment]
@Alex-302 

```adblock
[$domain=/exampl[exyz]+\.com)/]##body
!                           ↑ unopened parenthesis
[$domain=/example\.(com|org)/]##body
!                      ↑ pipe should be escaped
```

**Comment by Alex-302** [comment]
@scripthunter7 Isn't it separators check fail?

**Comment by Alex-302** [comment]
@scripthunter7 `|` is a part of regexp. Regexp with escaped `$`, `|`, `,` is not working in any other app. But this is required for AdGuard.



> We don't yet support this syntax

I just tell about regexp, which works in UBO without "redundant" escaping.

**Comment by scripthunter7** [comment]
@Alex-302 Yes, we'll need a special tokenization that allows regexps to be used without such escape constraints. But for this, the operation of the network rule / modifier parser must be completely reworked, several edge cases must be handled, care must be taken to ensure that no new issues arise and the parsing speed must also be kept at an appropriate level, but regardless, this improvement is already planned, and there are already related issues in the tracker, I think we'll implement it soon

**Comment by Alex-302** [comment]
Great)

**Comment by scripthunter7** [comment]
Duplicate of https://github.com/AdguardTeam/tsurlfilter/issues/121
